test/unit/org/apache/cassandra/db/commitlog/CommitLogSegmentManagerCDCTest.java

```diff
@@ -157,11 +157,25 @@ public void testCDCIndexFileWriteOnSync() throws IOException
         Assert.assertTrue("Index file not written: " + cdcIndexFile, cdcIndexFile.exists());
 
         // Read index value and confirm it's == end from last sync
-        BufferedReader in = new BufferedReader(new FileReader(cdcIndexFile));
-        String input = in.readLine();
-        Integer offset = Integer.parseInt(input);
-        Assert.assertEquals(syncOffset, (long)offset);
-        in.close();
+        String input = null;
+        // There could be a race between index file update (truncate & write) and read. See CASSANDRA-17416
+        // It is possible to read an empty line. In this case, re-try at most 5 times.
+        for (int i = 0; input == null && i < 5; i++)
+        {
+            try (BufferedReader in = new BufferedReader(new FileReader(cdcIndexFile)))
```

Comment:
Worth putting a small sleep in here to give the system the wiggle room to allow the scheduler to put the index file writing thread back on the stack and write the file out instead of just spinning on it 5 times?